### ISSUE TITLE
Forcing image aspect ratio to align with terminal grid

### ISSUE THREAD

**Issue by benlubas** [3rd/image.nvim#22]
Continuing conversation from #20. 

I've also noticed that a little. Although it's much less noticable on larger images. 

My thought here would be to just break out of the terminal grid. Allow images to be displayed but not take up the full height of a cell. 

I'm not too familiar with image protocols, but I'd imagine they're able to do this. If they can't (maybe sixels for example) we could probably pad an edge of the image with transparent pixels.

**Comment by 3rd** [comment]
Yep, we'd need to patch them with a transparent padding (https://sw.kovidgoyal.net/kitty/graphics-protocol/#controlling-displayed-image-layout), but the magick rock doesn't seem to support that, or at least I couldn't find a way to make it work :|

It's also pretty darn slow at everything, thinking of shipping an alternate CLI tool.
We only just convert from a format to another, scale and crop the images.
Such a tool could also add the padding, but it's gonna take some time, and I'm not sure if people will be willing to install it.

**Comment by benlubas** [comment]
I don't see anything on that page that suggests the image must conform to the terminal grid.

> You can also specify extra X=3 and Y=4 pixel offsets to display from a different origin within the cell.

This actually seems to indicate that they don't have to be.

Currently we're cropping the image manually to exactly line up with the grid. I wonder what happens if we just don't do that, although you might have already tried that and it might break horribly.

**Comment by glyh** [comment]
Hello, given https://github.com/kovidgoyal/kitty/issues/7380 this is closed, I think we don't need to rescale the image for now as kitty natively support it? I haven't test it though.

**Comment by 3rd** [comment]
I think we still need it, we need to convert from other image formats to PNG and also support ueberzug. In the rewrite I'm adding a magick cli-only mode to work around the rock being not good. Personally I don't like having to use imagemagick at all and I'd like to have a separate utility that does all image processing in rust/go, but opinions vary https://github.com/3rd/image.nvim/issues/124

**Comment by glyh** [comment]
You don't actually need to do the conversion. Kitty's image protocol accepts matricies of pixels.

**Comment by 3rd** [comment]
We'd have to read the pixels for all the supported image formats in Lua, which wouldn't be feasible, but even if we did have a way to do it, it would probably be slower than asking Kitty to read the png file directly. And we need to support ueberzug as well.
